cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/parser/MetadataParser.java

```diff
@@ -28,40 +28,109 @@
 @Component
 public class MetadataParser {
 
-    public static final String PREFIX = "##";
+    public static final String PREFIX = "#!";
     public static final int BEGIN_INDEX = 3;
+    public static final String BLOCK_END = "#!!#";
+    public static final String BLOCK_START = "#!!";
+    public static final String COLON = ":";
+    public static final List<DescriptionTag> DESCRIPTION_TAGS_LIST = Collections.unmodifiableList(
+            Arrays.asList(DescriptionTag.values()));
 
     @Autowired
-    private Yaml yaml;
-    @Autowired
     private ParserExceptionHandler parserExceptionHandler;
 
-    public ParsedMetadata parse(SlangSource source) {
+    public Map<String, String> parse(SlangSource source) {
         Validate.notEmpty(source.getSource(), "Source " + source.getName() + " cannot be empty");
-
-        String descriptionYAML = extractDescriptionYAMLString(source);
         try {
-            return yaml.loadAs(descriptionYAML, ParsedMetadata.class);
+            String fullDescription = extractFullDescriptionString(source);
+            Map<String, String> fullMap = extractFullTagMap(fullDescription);
+            checkMapOrder(fullMap);
+            return fullMap;
         } catch (Throwable e) {
-            throw new RuntimeException("There was a problem parsing the description in YAML format: " +
+            throw new RuntimeException("There was a problem parsing the description: " +
                     source.getName() + ".\n" + parserExceptionHandler.getErrorMessage(e), e);
         }
     }
 
-    private String extractDescriptionYAMLString(SlangSource source) {
+    private void checkMapOrder(Map<String, String> fullMap) {
```

Comment:
don't we want the users to keep an order in the description? @description, @prerequisites(optional), @input(optional), @output(optional), @result(optional)
do we want to let users mix inputs with outputs etc.?